Mage/src/main/java/mage/abilities/common/DealsDamageTriggeredAbility.java

```diff
@@ -0,0 +1,42 @@
+package mage.abilities.common;
+
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.effects.Effect;
+import mage.constants.Zone;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+
+/**
+ * @author LevelX2
+ */
+
+public class DealsDamageTriggeredAbility extends TriggeredAbilityImpl {
+
+    public DealsDamageTriggeredAbility(Effect effect) {
+        super(Zone.BATTLEFIELD, effect, false);
+        setTriggerPhrase("Whenever {this} deals damage, ");
+    }
+
+    protected DealsDamageTriggeredAbility(final DealsDamageTriggeredAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public DealsDamageTriggeredAbility copy() {
+        return new DealsDamageTriggeredAbility(this);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.DAMAGED_PERMANENT
+                || event.getType() == GameEvent.EventType.DAMAGED_PLAYER;
```

Comment:
(todo: fix per #11985 )

Reply:
Just want to clarify, is there a fix here or is it something getting tracked by 11985?

Reply:
it needs to be fixed but depends on that PR, so don't worry about it, I'll get it taken care of